packages/dds/quorum/src/quorum.ts

```diff
@@ -269,35 +271,19 @@ export class Quorum extends SharedObject<IQuorumEvents> implements IQuorum {
      */
     public delete(key: string): void {
         const currentValue = this.values.get(key);
-        // Early-exit if we can't submit a valid proposal (there's nothing to delete or already a pending proposal).
-        if (currentValue === undefined || currentValue.pending !== undefined) {
-            return;
-        }
-
-        // If not attached, we basically pretend we got an ack immediately.
-        // TODO: Should we just directly store the value rather than the full simulation?
-        if (!this.isAttached()) {
-            // Queueing as a microtask to permit callers to complete their callstacks before the result of the delete
-            // takes effect.  This more closely resembles the pattern in the attached state, where the ack will not
-            // be received synchronously.
-            queueMicrotask(() => {
-                this.handleIncomingDelete(
-                    key,
-                    0 /* refSeq */,
-                    0 /* setSequenceNumber */,
-                    "detachedClient" /* clientId */,
-                );
-            });
+        // Early-exit if:
+        if (
+            // there's nothing to delete
+            currentValue === undefined
+            // if something is pending (and so our proposal won't be valid)
+            || currentValue.pending !== undefined
```

Comment:
should this case be an error? how is it handled in set?

Reply:
I don't think there's a need, I think of this portion as being more of a performance optimization to avoid submitting useless ops.  It has the same motivation as the early-exit for set(), but delete() can be a little more aggressive since it's less ambiguous.

I think throwing for just these cases is not particularly helpful because even if we proceed past here and submit an op, there's no guarantee it will be valid by the time it was sequenced (if someone else beats us).  So the better pattern is for the caller to (1) verify they actually want to change the current value (i.e. don't delete undefined) (2) if so, call set()/delete() and (3) watch for the accept, which may be for a competing client and not your own.

To go over all the options I see:
1. What I'm doing here - minimal useless ops, but calls to delete() may not always be followed by an "accept" event if you blindly deleted something that didn't exist.
2. Match the early-exit for set() exactly -- deleting an undefined value still submits an op even though it's kind of a no-op.  Advantage is that then you're guaranteed that calls to delete() will always be followed by an "accept" event even if you delete something that's not there.  This would "lock" the key more though since further edits won't be possible until the delete is accepted.
3. Throw error for delete() of undefined -- minimal useless ops, but adds an error handling requirement to the caller which is no simpler than just checking the value before calling delete().
4. Throw error for any futile set()/delete() -- minimal useless ops, but also adds an error handling requirement or else they need to inspect both the current value and the pending value, which would be more overhead for them.